core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/variables/runtime/VariableContext.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.accesscontrol.blocks.variables.runtime
+
+import cats.data.NonEmptyList
+import tech.beshu.ror.accesscontrol.blocks.rules.Rule
+import tech.beshu.ror.accesscontrol.blocks.variables.runtime.MultiExtractable.SingleExtractableWrapper
+
+object VariableContext {
+
+  sealed trait VariableType
+
+  object VariableType {
+    trait User extends VariableType
+    trait CurrentGroup extends VariableType
+    trait Header extends VariableType
+    trait Jwt extends VariableType
+  }
+
+  trait UsingVariable {
+    this: Rule =>
+    def usedVariables: NonEmptyList[RuntimeResolvableVariable[_]]
+
+    def extractVariablesTypesFromExtractables: List[VariableType] = {
+      usedVariables.toList
+        .flatMap {
+          case RuntimeSingleResolvableVariable.ToBeResolved(extractables) => extractSingle(extractables)
+          case RuntimeMultiResolvableVariable.ToBeResolved(extractables) => extractMulti(extractables)
+          case _ => List.empty
+        }
+    }
+
+    private def extractSingle(extractables: NonEmptyList[SingleExtractable]) =
+      extractables.collect { case e: VariableContext.VariableType => e }
```

Comment:
do we have to use collect? Do we want to filter sth? Maybe we can use just map - it's way safer when you want to add sth in future

Reply:
We want use only variables, so for example `Const` should not be collected.